src/tools/yarprobotstatepublisher/src/robotstatepublisher.cpp

```diff
@@ -62,13 +62,14 @@ YARPRobotStatePublisherModule::YARPRobotStatePublisherModule(): m_iframetrans(nu
 bool YARPRobotStatePublisherModule::configure(ResourceFinder &rf)
 {
     string name="yarprobotstatepublisher";
-    m_rosNode = new yarp::os::Node("/yarprobotstatepublisher");
+    string namePrefix = rf.check("namePrefix",Value("")).asString();
+    m_rosNode = new yarp::os::Node("/"+namePrefix+"/yarprobotstatepublisher");
```

Comment:
Have you considered if this change in the Node name can be problematic for back-compatibility? Furthermore, if `namePrefix` is empty, it is not a big strange to have the Node to be named `//yarprobotstatepublisher`?

Reply:
My bad, I did not consider the back compatibility!
Also ideally it is not good to have a ros node with the name `//yarprobotstatepublisher`.
I will update this logic to account for back compatibility